src/index.ts

```diff
@@ -3,14 +3,30 @@ import {
     TransactContext,
     TransactHookResponseType,
     TransactHookTypes,
+    Transaction,
 } from '@wharfkit/session'
 
 /** Import JSON localization strings */
 import defaultTranslations from './translations.json'
 
-export class TransactPluginTemplate extends AbstractTransactPlugin {
+const START_CHECKING_FINALITY_AFTER = 150000 // 2.5 minutes
+
+interface TransactPluginFinalityCallbackOptions {
+    onFinalityCallback: () => void
```

Comment:
The callback function should return data, probably the data from the API call, so the callback can work with that data. Right now the callback only knows the transaction is final, but maybe there's some data in from that API call that'd be helpful to them?